mujoco_warp/_src/collision_primitive.py

```diff
@@ -232,6 +232,44 @@ def plane_box(
       break
 
 
+@wp.func
+def sphere_cylinder(
```

Comment:
This is surprisingly simple compared to C MuJoCo's Sphere<>Cylinder:

https://github.com/google-deepmind/mujoco/blob/main/src/engine/engine_collision_primitive.c#L324

Does this handle all three cases that are explicitly laid out in C MuJoCo: side, cap, corner?

If so, cool!  Perhaps C MuJoCo ought to use this code!  cc @yuvaltassa

If not, please fix.

Reply:
Ya I think this code is missing some cases. The point of side/cap/corner is that the normals are always aligned correctly.

In particular `proj = wp.clamp(wp.dot(delta, axis), -cylinder.size[1], cylinder.size[1])` is what you want for cap and corner but not for side. In the current implementation for the side collision case `closest` will not actually be the closest point to the sphere center.

The function in engine_collision_primitive is well tested. Happy to write simple test model for you covering all the branches if this would help.